modules/Users/User.php

```diff
@@ -1532,7 +1532,7 @@ public function getEmailLink(
             $emailLink = $emailUI->populateComposeViewFields($focus);
         } else {
             // straight mailto:
-            $emailLink = '<a href="mailto:' . $focus->$attribute . '" class="' . $class . '">';
+            $emailLink = sprintf('<a href="mailto:%1$s">%1$s</a>', $focus->$attribute);
```

Comment:
You have removed the CSS class attribute. Is this intential. It may effect the confirmed opt in logic.

Reply:
Yes, it was intentinally, since this method's default class value is an empty string, and there is no method exists which calls this method and set the class parameter. There is no effect for confirmed opt in logic...

Reply:
OK. I think we should take a note of this for the future. Perhaps we should remove the parameters? $contact_id, $ret_action, $ret_id and $class have no effect on this method. Or we could update the PHP Doc to state that the params are deprecated.

Reply:
Ye, we could remove unnecessary params easily. Gonna create a new ticket for refactoring.